examples/22_define-router/src/entries.tsx

```diff
@@ -193,4 +193,24 @@ export default defineRouter({
     }
     throw new Error('renderRoute: No such path:' + path);
   },
+  getApiConfig: async () => [
+    {
+      path: [
+        { type: 'literal', name: 'api' },
+        { type: 'literal', name: 'hi' },
+      ],
+      mode: 'static',
+      method: 'GET',
+      handler: async () => {
+        return new Response('Hello World');
+      },
+    },
+  ],
+  handleApi: async (path) => {
+    console.log('api called', path);
+    return {
+      headers: {},
+      status: 200,
+    };
```

Comment:
Though, this looks valid and I expect it returns an empty content. Not sure where it returns an html.

Reply:
it's just when body is undefined or null 👀 - I can look into it

Reply:
I found a place where we can set the body in the middleware, but not sure if that is the right place to fix this or not. Let me know what you think of the change and where may be better to fix it if that's the case

Reply:
`body` is optional, so you don't need to set it. Or, I misunderstand your question.

Reply:
This change in the middleware handler: https://github.com/dai-shi/waku/pull/1116/files#diff-4142accf18d84717732f9aa0ed86505383a968a3369c6049dea5c620454322e0L157-L160

is to fix the issue in the description: https://github.com/dai-shi/waku/pull/1116#issue-2766388289

the e2e tests confirm that it is the wrong fix, but I think there is a case where we should return an empty body from the server and instead we send html still.

Reply:
Yeah, that's a wrong fix. It's not the handler job to create the body. Maybe, I misunderstand the hono behavior.

Reply:
Can you check if `c.body(null)` is causing the problem?

https://github.com/dai-shi/waku/blob/52f7ad61ad456df65db3221407cca810c95d7646/packages/waku/src/lib/hono/engine.ts#L64

Reply:
It isn't this^. 

Somehow the middleware handler is setting the `body` to a readablestream (it should just not be defined on the ctx object I think), but I can't see where it is making the change yet.

Reply:
looking into it. it looks like it's vite behavior.

Reply:
so, there were various bugs, most of which are my bad.